File: android/src/main/java/org/ligi/gobandroid_hd/logic/StatelessBoardCell.java
package org.ligi.gobandroid_hd.logic;

import java.util.ArrayList;
import java.util.List;

public class StatelessBoardCell extends Cell {

    public final StatelessGoBoard board;

    private List<StatelessBoardCell> neighbours = new ArrayList<>();
    public StatelessBoardCell left, right, up, down;


    public StatelessBoardCell(int x, int y,StatelessGoBoard board) {
        super(x, y);
        this.board = board;
    }

    public void assignNeighbours() {
        if (x > 0) {
            left = board.getCell(x - 1, y);
            neighbours.add(left);
        }

        if (y > 0) {
            up = board.getCell(x, y - 1);
            neighbours.add(up);
        }

        if (y < board.getSize() - 1) {
            down = board.getCell(x, y + 1);
            neighbours.add(down);
        }


        if (x < board.getSize() - 1) {
            right = board.getCell(x + 1, y);
            neighbours.add(right);
        }
    }



    public List<StatelessBoardCell> getNeighbors() {
        return neighbours;
    }
}

File: android/src/main/java/org/ligi/gobandroid_hd/logic/StatelessBoardCell.kt
package org.ligi.gobandroid_hd.logic

import java.util.*

class StatelessBoardCell(val cell: CellImpl, val board: StatelessGoBoard) : Cell by cell {

    var left: StatelessBoardCell? = null
    var right: StatelessBoardCell? = null
    var up: StatelessBoardCell? = null
    var down: StatelessBoardCell? = null

    val neighbors = HashSet<StatelessBoardCell>()

    fun assignNeighbours() {
        if (x > 0) {
            left = board.getCell(x - 1, y)
            neighbors.add(left!!)
        }

        if (y > 0) {
            up = board.getCell(x, y - 1)
            neighbors.add(up!!)
        }

        if (y < board.size - 1) {
            down = board.getCell(x, y + 1)
            neighbors.add(down!!)
        }

        if (x < board.size - 1) {
            right = board.getCell(x + 1, y)
            neighbors.add(right!!)
        }
    }
}
